com.wamas.ide.launching/src/com/wamas/ide/launching/generator/BundleFilter.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) SSI Schaefer IT Solutions
+ */
+package com.wamas.ide.launching.generator;
+
+import java.util.Collection;
+import java.util.function.Predicate;
+
+import org.eclipse.osgi.service.resolver.BundleDescription;
+
+public interface BundleFilter extends Predicate<BundleDescription> {
+
+    public static class Factory {
```

Comment:
the whole Factory feels like overkill, as everything here can be archived with java.util.function